From 8b5d9e76eecb09e9f6725cf0c2c8a2e2b0b4d9d1 Mon Sep 17 00:00:00 2001
From: Rezoss <rezoss@local>
Date: Sun, 7 Jun 2026 11:20:00 +0200
Subject: [PATCH] Force enable Apple AirDrop ProtocolX support

ShareLive already contains the Apple-device / AirDrop ProtocolX UI and
runtime paths on the current S23U base. Force the local Samsung-side
ProtocolX gate on and prevent policy sync from clearing the Apple-device
sharing option. Google Play services 26.11.xx+ is still required for the
GMS/Nearby backend.

diff --git a/smali/vf/f5.smali b/smali/vf/f5.smali
index 76900b9..25a9f30 100644
--- a/smali/vf/f5.smali
+++ b/smali/vf/f5.smali
@@ -274,91 +274,11 @@
 .end method
 
 .method public final e(Z)Llw/u;
     .locals 3
 
-    sget-object v0, Lee/d;->C:Lee/d;
-
-    iget-boolean v0, v0, Lee/d;->m:Z
-
-    if-eqz v0, :cond_0
-
-    iget-object v0, p0, Lvf/f5;->g:Lpe/b2;
-
-    iget-object v0, v0, Lpe/b2;->n:Ljava/lang/Object;
-
-    check-cast v0, Landroid/content/SharedPreferences;
-
-    const-string v1, "protocol_x_supported"
-
-    const/4 v2, 0x1
-
-    invoke-interface {v0, v1, v2}, Landroid/content/SharedPreferences;->getBoolean(Ljava/lang/String;Z)Z
-
-    move-result v0
-
-    if-eqz v0, :cond_0
-
-    new-instance v0, Lsh/a;
-
-    const/16 v1, 0x11
-
-    invoke-direct {v0, v1, p0}, Lsh/a;-><init>(ILjava/lang/Object;)V
-
-    new-instance v1, Lax/b;
-
-    const/4 v2, 0x3
-
-    invoke-direct {v1, v2, v0}, Lax/b;-><init>(ILjava/lang/Object;)V
-
-    iget-object p0, p0, Lvf/f5;->c:Lxf/e;
-
-    check-cast p0, Lrf/m0;
-
-    invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
-
-    new-instance v0, Lrf/k0;
-
-    const/16 v2, 0xa
-
-    invoke-direct {v0, p0, v2}, Lrf/k0;-><init>(Lrf/m0;I)V
-
-    new-instance p0, Lax/b;
-
-    const/4 v2, 0x3
-
-    invoke-direct {p0, v2, v0}, Lax/b;-><init>(ILjava/lang/Object;)V
-
-    sget-object v0, Lrf/t;->y:Lrf/t;
-
-    invoke-virtual {p0, v0}, Llw/u;->e(Low/e;)Lax/g;
-
-    move-result-object p0
-
-    new-instance v0, Lb6/a;
-
-    const/16 v2, 0xa
-
-    invoke-direct {v0, p1, v2}, Lb6/a;-><init>(ZI)V
-
-    invoke-static {v1, p0, v0}, Llw/u;->x(Llw/u;Llw/u;Low/b;)Llw/u;
-
-    move-result-object p0
-
-    return-object p0
-
-    :cond_0
-    if-eqz p1, :cond_1
-
-    sget-object p0, Lxc/g;->B:Lxc/e;
-
-    const-string p1, "[ProtocolXRepository] supportsProtocolX: policyPao.protocolXSupported=false"
-
-    invoke-virtual {p0, p1}, Lxc/g;->e(Ljava/lang/String;)V
-
-    :cond_1
-    sget-object p0, Ljava/lang/Boolean;->FALSE:Ljava/lang/Boolean;
+    sget-object p0, Ljava/lang/Boolean;->TRUE:Ljava/lang/Boolean;
 
     invoke-static {p0}, Llw/u;->i(Ljava/lang/Object;)Lax/b;
 
     move-result-object p0
 
diff --git a/smali/vf/r3.smali b/smali/vf/r3.smali
index d736816..50abbcc 100644
--- a/smali/vf/r3.smali
+++ b/smali/vf/r3.smali
@@ -516,11 +516,13 @@
 
     const-string v3, "protocol_x_supported"
 
+    const/4 v2, 0x1
+
     invoke-interface {v1, v3, v2}, Landroid/content/SharedPreferences$Editor;->putBoolean(Ljava/lang/String;Z)Landroid/content/SharedPreferences$Editor;
 
     invoke-interface {v1}, Landroid/content/SharedPreferences$Editor;->apply()V
 
-    if-nez v2, :cond_d
+    goto :cond_d
 
     iget-object v0, v0, Lvf/x3;->e:Lkf/m;
 
